### PR TITLE
Bug: adding release or deployment ID to the status report

### PR DESCRIPTION
**Pull request by izaaklauer**
I don't think this is the *right* way to wire this information through, but it seems to work for now.

### CODE REVIEW

File: internal/core/app_status_report.go [hunk 1]
```diff
@@ -333,6 +334,16 @@ func (op *statusReportOperation) Do(
 	}
 	realMsg.ResourcesHealth = resourcesHealth
 
+	// Add the deployment/release ID to the report. TODO: wire the resource ID through explicitly
+	switch op.Target.(type) {
```
Comment (mitchellh) [inline internal/core/app_status_report.go:338]: You want to use a type switch with an assignment here: https://golang.org/doc/effective_go#type_switch

That'll clean things up a lot for you.

**Review by briancain — CHANGES_REQUESTED** [review]
Just a quick change on the error return!

**Comment by mitchellh** [comment]
Also @briancain is there no way to test this? This seems like a good test candidate.

**Comment by briancain** [comment]
@mitchellh Yep, we should update the tests too for this and be more explicit as to what values are returned. Right now we're mostly just checking that something was returned: 

https://github.com/hashicorp/waypoint/blob/main/internal/core/app_status_report_test.go#L124-L126

If you don't mind @izaaklauer - we should also add a check to match deploy/release id in the report.

**Review by briancain — APPROVED** [review]

**Pull request merged**